PCSX2/pcsx2
# SCP XInput Bridge Integration

### Description of Changes
DsHidMini, the newest driver for DualShock 3 gamepads, comes with a [SCP XInput Bridge DLL](https://vigem.org/projects/DsHidMini/SCP-XInput-Bridge/), implementing a SCP-era extension of the XInput API. With this extension, button pressure of DS3 gamepads may be read by a supported application. PCSX2 used to support this API extension during lilypad days (AFAIK), but now it was gone. This PR brings it back as a part of the XInput backend, and kicks in automatically once DsHidMini's `xinput1_3.dll` is placed in the PCSX2 directory.

This PR also fixes wrong triggers scaling for "normal" XInput.

### Rationale behind Changes
Pressure sensitivity is sometimes important, and SCP XInput Bridge is likely the cleanest way to obtain it. Plus, it used to be supported.

### Suggested Testing Steps
1. Dust off, buy, steal and/or 3D print a DualShock 3.
2. Install [DsHidMini](https://vigem.org/projects/DsHidMini/How-to-Install/).
3. Put DsHidMini's `xinput1_3.dll` in PCSX2 directory.
4. Launch PCSX2 and toggle Enable XInput Source. Ensure the gamepad is mapped properly in XInput mode.
5. Launch any application that you can verify pressure sensitive input data, e.g. racing games or `padtest`.
6. Verify that pressure sensitivity works as intended.

![pressure](https://user-images.githubusercontent.com/7947461/163257833-ff210385-e406-409e-b483-bf196af08fc7.jpg)
